### ISSUE TITLE
Relay Dimmer Issue

### ISSUE THREAD

**Issue by sidharthb0** [markbegma/homebridge-hdl-buspro#6]
Thanks for this awesome plug-in. 

I’ve been using the relay switches and dry contact sensors for a while now and all seems to be working well.

I’m unable to figure out the exact configuration for relay dimmers. I’ve configured it as per the setup tool but dimmers don’t seem to be working. 

![DAE7D83E-4185-446A-B2B0-6261B741B191](https://user-images.githubusercontent.com/23515848/203236565-03fabe6e-2c1d-478c-9c26-33a937184d83.jpeg)
![2978E1AC-756A-4D1A-940E-493FCFF32912](https://user-images.githubusercontent.com/23515848/203236575-62590c9d-6bc4-49ef-a031-38096885390b.jpeg)
![289DF993-CE7A-47CC-90DD-381691B014B2](https://user-images.githubusercontent.com/23515848/203236579-310cad00-d0f6-4c14-994c-80ba305a476c.jpeg)

My config is as per the images above. For the dimmer “Living Track”, as per the setup tool it’s Subnet: 1; Address: 1; Channel: 68. 

Using the same method for switches works but dimmers aren’t working. Any advice on how this should be configured?

**Comment by markbegma** [comment]
Hello @sidharthb0!

The config seems fine to me, though it has extra fields but they should not affect the device. Could you specify what do you mean by dimmers not working? is there no button? they don't light up? can't regulate dimming levels?
As long as lights are controlled from the same HDL bus it should not matter what is the producer.

**Comment by sidharthb0** [comment]
> Hello @sidharthb0!
> 
> The config seems fine to me, though it has extra fields but they should not affect the device. Could you specify what do you mean by dimmers not working? is there no button? they don't light up? can't regulate dimming levels? As long as lights are controlled from the same HDL bus it should not matter what is the producer.

Hey @markbegma thanks for replying. 

The buttons show up in the Home app along with dimmer controls but when I use those buttons there’s no feedback from the lights. When I use the relay switches, I also hear a ‘click’ sound in my HDL bus box but I don’t hear this when I’m using the dimmer switches. 

Almost seems like one of the Subnet - ID - Channel numbers is wrong, but I’ve followed the numbers in the bus setup tool.

**Comment by markbegma** [comment]
Checked for what the DALI is and it seems to be among the newer relays. The plugin was coded according to an old leaked internal instruction, so there is a chance it is not supported. But I also checked the manual for it and it seems that channels 65 to 80 virtually correspond to the entire groups of light, and i never had in mind how would this config work. https://b2b.hdl-automation.cz/en/support/products/manuals/buspro/dali/m-hdl-mc64-dali-431(dali-controller) Could you rather try to write in one separate dimmer from this group, maybe it would work?

<img width="1250" alt="Снимок экрана 2022-11-23 в 12 52 12281" src="https://user-images.githubusercontent.com/64325270/203505012-0f0adcbc-5695-4a43-b191-e7fe76d3e23f.png">

**Comment by sidharthb0** [comment]
That makes sense. 

Do you mean to try and use DALi channel 0 instead of HDL channel 65?

**Comment by markbegma** [comment]
No, I rather mean one of HDL channels 1-64, that are occupied by the system, DALI config system should have which are used.

**Comment by sidharthb0** [comment]
Alright let me try that and report back. Thank you.

**Comment by sidharthb0** [comment]
Reporting back - unfortunately I wasn’t able to find any channel that worked. What I did find though was that when I use the physical touchpad on my wall to turn on a relay dimmable light, it reflected immediately on the Home app on my phone. 

Any dimming using the physical touchpad or HDL app also reflects in the Home app with my original configuration. It seems that home app is picking up current state but is not able to send commands to the bus. Weird. 

Any idea why this could be happening?

**Comment by markbegma** [comment]
> Reporting back - unfortunately I wasn’t able to find any channel that worked. What I did find though was that when I use the physical touchpad on my wall to turn on a relay dimmable light, it reflected immediately on the Home app on my phone.
> 
> Any dimming using the physical touchpad or HDL app also reflects in the Home app with my original configuration. It seems that home app is picking up current state but is not able to send commands to the bus. Weird.
> 
> Any idea why this could be happening?

I assume that DALI must have a different bytestring payload to command the lights compared to the usual [single channel command](https://github.com/caligo-mentis/smart-bus#complete-example), maybe something like the scene control command, while the listener payload coincides with the single channel one. This is weird since the manual for DALI states that 65-80 accept single channel commands to control the groups. I would have expected that it should behave like an adapter, but you seem to have troubles with this one. Maybe with DALI we need to code in area to the contents section of the string (doc attached).
Honestly have trouble helping you with this one... I don't have DALI devices and moreover - won't be at home with my HDL setup to tinker with it for quite a while. You can try to send different commands with nodejs, with template from smart-bus repository above. If you find the format that will turn on the dimmer, I might code it into the plugin the blind way...

**Comment by sidharthb0** [comment]
Thanks and apologies for the delayed response. Let me tinker around with this and hopefully I’ll come across something that works. 

Appreciate your help and patience.

**Comment by sidharthb0** [comment]
Updating for posterity - I used a Home Assistant plugin to resolve this issue for relay dimmable lights.

https://github.com/eyesoft/home_assistant_buspro/
